addons/test_mass_mailing/tests/test_mailing_internals.py

```diff
@@ -120,3 +120,21 @@ def test_mailing_trace_utm(self):
             self.assertEqual(test_utm.campaign_id, campaign)
             self.assertEqual(test_utm.source_id, source)
             self.assertEqual(test_utm.medium_id, medium)
+
+    def test_mailing_test_button(self):
+        mailing = self.env['mailing.mailing'].create({
+            'name': 'TestButton',
+            'subject': 'Test Subject',
+            'state': 'draft',
+            'mailing_type': 'mail',
+            'body_html': '<p>Hello ${object.name}</p>',
+            'mailing_model_id': self.env['ir.model']._get('res.partner').id,
+        })
+        mailing_test = self.env['mailing.mailing.test'].create({
+            'email_to': 'test@test.com',
+            'mass_mailing_id': mailing.id,
+        })
+        mailing_test.send_mail_test()
+        mailing.write({'body_html': '<p>Hello ${object.name_id.id}</p>'})
```

Comment:
Just a small comment to tell that you test bad jinja ? Could be great also to check each corner case: bad body, bad subject, bad preview. So that we cover all cases...